tiledb/sm/query/query_buffer.h

```diff
@@ -239,10 +239,19 @@ class QueryBuffer {
     return Status::Ok();
   }
 
-  /** Returns a const pointer to the buffer data as the requested type. */
+  /** Returns a const pointer to the data buffer as the requested type. */
   template <typename T>
-  inline const T* typed_buffer() const {
-    return buffer_ ? static_cast<T*>(buffer_) : nullptr;
+  inline const T* typed_data_buffer() const {
```

Comment:
I think we used the "convention" of data_buffer_as in other places?

Reply:
I was copying the name from the `Range::typed_data` method. Should I update that name as well?